app-connector-client/server/routes/events.js

```diff
@@ -48,67 +50,93 @@ async function createEventsFromConfig(localKyma, eventsConfig, registeredApis) {
+                reject(err)
+            });
 
     })
 }
-function updateEvent(localKyma, eventMetadata, event, event_id) {
+async function updateEvent(localKyma, eventMetadata, event, event_id) {
     LOGGER.debug("Auto-update Event API '%s'", event.name)
     return new Promise((resolve, reject) => {
-        eventMetadata = fillEventData(eventMetadata, event);
-        apis.updateAPI(localKyma, eventMetadata, event_id, function (err, httpResponse, body) {
-            if (err) {
-                reject(err)
-            } else {
-                if (httpResponse.statusCode >= 400) {
-                    var err = new Error(body.error);
-                    reject(err);
+        fillEventData(eventMetadata, event).then(function (result) {
+            eventMetadata = result;
+            apis.updateAPI(localKyma, eventMetadata, event_id, function (err, httpResponse, body) {
+                if (err) {
+                    reject(err)
+                } else {
+                    if (httpResponse.statusCode >= 400) {
+                        var err = new Error(body.error);
+                        reject(err);
+                    }
+                    resolve(body)
                 }
-                resolve(body)
-            }
-        })
-
+            })
+        }).catch(function (err) {
+            reject(err)
+        });
     })
 }
 function fillEventData(eventMetadata, event) {
-    eventMetadata.name = event.name;
-    if (eventMetadata.description) {
-        eventMetadata.description = event.description;
-    }
-    else {
-        eventMetadata.description = event.name;
-    }
-    if (eventMetadata.labels) {
-        eventMetadata.labels = event.labels;
-    }
+    return new Promise(function (resolve, reject) {
+        eventMetadata.name = event.name;
+        if (eventMetadata.description) {
+            eventMetadata.description = event.description;
+        }
+        else {
+            eventMetadata.description = event.name;
+        }
+        if (eventMetadata.labels) {
+            eventMetadata.labels = event.labels;
+        }
 
-    var specInJson
-    if (event.specification.endsWith(".json")) {
-        specInJson = JSON.parse(fs.readFileSync(event.specification))
-    } else {
-        specInJson = yaml.safeLoad(fs.readFileSync(event.specification, 'utf8'));
-    }
+        var specInJson
+        if (event.specification.endsWith(".json")) {
+            specInJson = JSON.parse(fs.readFileSync(event.specification))
+        } else {
+            specInJson = yaml.safeLoad(fs.readFileSync(event.specification, 'utf8'));
+        }
+        check_api.check_api(specInJson)
+        refParser.dereference(specInJson)
+            .then(function (schema) {
+                Object.keys(schema.topics).forEach((topicKey) => {
+                    if (schema.topics[topicKey].publish) {
+                        schema.topics[topicKey].example = openapiSampler.sample(schema.topics[topicKey].publish)

```

Comment:
so now you are editing the event api on server side at registration time by generating the example out of json schema and attaching it to the topic definition.

Why you are not doing in on client side? That will allow to render it as well for apis not registered by the mock?

What is if there is an example defined already?